### PR TITLE
Ben/add multichoice qa

### PR DESCRIPTION
**Pull request by benleetownsend**
Needs a proper dataset and some tests, but this is the basic idea.

### CODE REVIEW

File: finetune/multi_choice_qa.py [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+            formatted_predictions.append(
+                dict(zip(answers_per_sample, probas))
+            )
+        return formatted_predictions
+
+    def featurize(self, question, answers, max_length=None):
+        """
+        Embeds inputs in learned feature space. Can be called before or after calling :meth:`finetune`.
+
+        :param X1: List or array of text, shape [batch]
```
Comment (benleetownsend) [inline finetune/multi_choice_qa.py]: Docs need update... oops

File: finetune/multi_choice_qa.py [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+        kwargs['tokens'] = [arr.tokens for arr in arrays]
+        kwargs['token_ids'] = np.stack([arr.token_ids for arr in arrays], 1)
+        kwargs['mask'] = np.stack([arr.mask for arr in arrays], 1)
+        return ArrayEncodedOutput(**kwargs)
+
+    def finetune(self, question, correct_answer, other_answers, batch_size=None, fit_lm_only=False):
+        """
+        :param question: List or array of text, shape [batch]
+        :param correct_answer: List or array of correct answers [batch]
+        :param answers: List or array of text, shape [n_answers - 1 , batch]
```
Comment (madisonmay) [inline finetune/multi_choice_qa.py]: incorrect_answers

**Comment by madisonmay** [comment]
Test uncommitted

**Comment by madisonmay** [comment]
Resolves #85

**Comment by madisonmay** [comment]
Instead of correct_answers and incorrect_answers, take in question, answers, correct_answer

File: finetune/multiple_choice.py [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+        raw_probas = self._predict_proba(question, answers, max_length)
+
+        formatted_predictions = []
+        for probas, *answers_per_sample in zip(raw_probas, *answers):
+            formatted_predictions.append(
+                dict(zip(answers_per_sample, probas))
+            )
+        return formatted_predictions
+
+    def featurize(self, question, answers, max_length=None):
```
Comment (madisonmay) [inline finetune/multiple_choice.py]: This is kinda odd.  Not the behavior I would expect for featurize.

File: finetune/multiple_choice.py [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+        Format multi question examples as a list of IDs
+        """
+        arrays = [super(MultipleChoice, self)._text_to_ids(question, ans) for ans in answers]
+        kwargs = arrays[0]._asdict()
+        kwargs['tokens'] = [arr.tokens for arr in arrays]
+        kwargs['token_ids'] = np.stack([arr.token_ids for arr in arrays], 1)
+        kwargs['mask'] = np.stack([arr.mask for arr in arrays], 1)
+        return ArrayEncodedOutput(**kwargs)
+
+    def finetune(self, question, correct_answer, answers, batch_size=None, fit_lm_only=False):
```
Comment (madisonmay) [inline finetune/multiple_choice.py]: Swap the order of answers and correct_answers (so it's like inputs, target) and this is good to ship.

**Pull request merged**